cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -173,15 +179,28 @@ private void setFollowerThrottledReplicas(String topic, Set<String> replicas) {
   }
 
   private void setThrottledReplicas(String topic, Set<String> replicas, String configKey) {
-    assert (configKey.equals(LEADER_THROTTLED_REPLICAS) || configKey.equals(FOLLOWER_THROTTLED_REPLICAS));
+    if (!configKey.equals(LEADER_THROTTLED_REPLICAS) && !configKey.equals(FOLLOWER_THROTTLED_REPLICAS)) {
+      throw new IllegalArgumentException(
+          String.format("Config key must be either %s or %s", LEADER_THROTTLED_REPLICAS, FOLLOWER_THROTTLED_REPLICAS));
+    }
+
     Properties config = _kafkaZkClient.getEntityConfigs(ConfigType.Topic(), topic);
-    // Merge new throttled replicas with existing configuration values.
-    Set<String> newThrottledReplicas = new TreeSet<>(replicas);
     String oldThrottledReplicas = config.getProperty(configKey);
-    if (oldThrottledReplicas != null) {
-      newThrottledReplicas.addAll(Arrays.asList(oldThrottledReplicas.split(",")));
+    if (oldThrottledReplicas != null && oldThrottledReplicas.trim().equals("*")) {
+      // The existing setup throttles all replica. So, nothing needs to be changed
```

Comment:
nit - end-of-line dot missing
```suggestion
      // The existing setup throttles all replica. So, nothing needs to be changed.
```